src/screens/Connectors/PaymentProcessor/ConnectorMetaData/PaySafe/PaySafe.res

```diff
@@ -0,0 +1,191 @@
+      let flattern = formState.values->JsonFlattenUtils.flattenObject(true)
+      let name = `metadata.account_id.${account}.${currency}.${config}`
+      switch flattern->Dict.get(name) {
+      | Some(_) => true
+      | None => false
+      }
+    }
+    let {globalUIConfig: {font: {textColor}}} = React.useContext(ThemeProvider.themeContext)
+    <div>
+      <div
+        className={"flex gap-1.5 items-center cursor-pointer"}
+        onClick={_ => handleShowConfigModal(~showModal=true)}>
+        <CheckBoxIcon isSelected={checkIsSelected()} />
+        <p className="text-sm font-medium leading-5 text-grey-700 opacity-50">
+          {config->snakeToTitle->React.string}
+        </p>
+      </div>
+      <RenderIf condition=showConfigModal>
+        <Modal
+          modalHeading="Additional Details to enable"
+          headerTextClass={`${textColor.primaryNormal} font-bold text-xl`}
+          headBgClass="sticky top-0 z-30 bg-white"
+          showModal={showConfigModal}
+          setShowModal={setShowAccountConfigModal}
+          onCloseClickCustomFun={() => handleShowConfigModal(~showModal=false)}
+          paddingClass=""
+          revealFrom=Reveal.Right
+          modalClass="w-full md:w-1/3 !h-full overflow-y-scroll !overflow-x-hidden rounded-none text-jp-gray-900"
+          childClass={""}>
+          <AccountIdCurrencyFields
+            configDict
+            selectedConfig=config
+            account
+            currency
+            handleShowConfigModal
+            setShowAccountConfigModal
+          />
+        </Modal>
+      </RenderIf>
+    </div>
+  }
+}
+
+module AccountIdMethod = {
+  @react.component
+  let make = (~currency, ~currencyDict, ~account) => {
+    open LogicUtils
+    let configDict = currencyDict->getDictfromDict(currency)
+    <div className="border border-nd_gray-150 rounded-xl overflow-hidden p-6 flex flex-col gap-4">
+      <p className="font-semibold text-bold text-lg"> {currency->snakeToTitle->React.string} </p>
+      <hr />
+      <div className="grid grid-cols-4 gap-2">
+        {configDict
+        ->Dict.keysToArray
+        ->Array.map(config => {
+          <AccountCheckBox key={randomString(~length=10)} config configDict account currency />
+        })
+        ->React.array}
+      </div>
+    </div>
+  }
+}
+
+module AccountId = {
+  @react.component
+  let make = (~accountIdDict, ~account) => {
+    open LogicUtils
+
+    let currencyDict =
+      accountIdDict->getDictfromDict(account)->JSON.Encode.object->convertMapObjectToDict
+
+    <div className="flex flex-col gap-6 col-span-3">
+      <div className="max-w-3xl flex flex-col gap-8">
+        {currencyDict
+        ->Dict.keysToArray
+        ->Array.map(currency => {
+          <AccountIdMethod key={randomString(~length=10)} currency currencyDict account />
+        })
+        ->React.array}
+      </div>
+    </div>
+  }
+}
+
+@react.component
+let make = (~connectorMetaDataFields) => {
+  open LogicUtils
```

Comment:
open LogicUtils once at the top of the file, instead of opening in each component